core/src/main/java/org/apache/calcite/plan/StrongPredicateAnalyzer.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.plan;
+
+import org.apache.calcite.rex.RexCall;
+import org.apache.calcite.rex.RexNode;
+
+import java.util.List;
+
+/**
+ * Utilities for checking if a predicate is definitely false based on a strong policy.
+ */
+public final class StrongPredicateAnalyzer {
```

Comment:
Do you really need to add a new util class? How about just adding these methods to `Strong` since we already have many similar methods.
Besides, adding methods to `Strong` makes Calcite more extensible, e.g., some downstream projects could extends `Strong` and override some methods to add their own logics.

Reply:
Thanks for the remark, done!